### DISCUSSION TITLE
Imported from url property in the files table.

### DISCUSSION THREAD

**Discussion by cory-glooh** [directus/directus#8895] [Archive]
We have a script that synchronises external data into our database. importing any files references into directus. to avoid duplicate files being imported, would be nice if there was a 'imported_from' property available in the directus_files table.

**Comment by rijkvanzanten** [comment]
You can add custom fields to the system collections from the data model page 🙂

**Reply by cory-glooh** [reply]
@rijkvanzanten This wouldn't be auto-populated when a file is imported into diectus though? using the files.import api

**Reply by rijkvanzanten** [reply]
That's correct, but you should be able to create a file upload hook (https://docs.directus.io/guides/api-hooks/) to pre-fill that field on uploads automatically 👍🏻

**Reply by cory-glooh** [reply]
@rijkvanzanten The file upload hook doesn't contain any information regarding the source of the image, only it's name. there would need to be a files.import hook, which exposes the importURL

**Reply by rijkvanzanten** [reply]
Ah, yeah that's correct

**Reply by cory-glooh** [reply]
@rijkvanzanten Can this be re-opened, or shall I work on a pull request for the files.import hook?

**Reply by rijkvanzanten** [reply]
This is not an issue that can be closed/opened, but yeah, feel free to go ahead on the files.import hook! Sounds like a useful (and obvious) thing to have 🙂

**Comment by hugopvl** [comment]
@cory-glooh 
> We have a script that synchronises external data into our database. importing any files references into directus. to avoid duplicate files being imported, would be nice if there was a 'imported_from' property available in the directus_files table.

I think I'll need something like that, is it possible to share this script :D
